src/components/People/PeopleTable.tsx

```diff
@@ -0,0 +1,190 @@
+    ));
+  }
+
+  if (sex !== null) {
+    displayedPeople = displayedPeople.filter(person => person.sex === sex);
+  }
+
+  switch (sort) {
+    case 'name':
+    case 'sex':
+      displayedPeople.sort((p1, p2) => {
+        return order === 'desc'
+          ? p2[sort].localeCompare(p1[sort])
+          : p1[sort].localeCompare(p2[sort]);
+      });
+      break;
+    case 'died':
+    case 'born':
+      displayedPeople.sort((p1, p2) => {
+        return order === 'desc' ? p2[sort] - p1[sort] : p1[sort] - p2[sort];
+      });
+      break;
+    default:
+      break;
+  }
+
+  const setSortParam = (sortParameter: string): SearchParams => {
+    if (sort === null || sort !== sortParameter) {
+      return { sort: sortParameter, order: null };
+    }
+
+    switch (order) {
+      default:
+      case null:
+        return { order: 'desc' };
+      case 'desc':
+        return { sort: null, order: null };
+    }
+  };
+
+  const setSortColumnClasses = (sortBy: string) => {
+    return classNames('fas', {
+      'fa-sort': sort !== sortBy,
+    }, {
+      'fa-sort-down': sort === sortBy && order === 'desc',
+    }, {
+      'fa-sort-up': sort === sortBy && order === null,
+    });
+  };
+
+  return (
+    <>
+      {displayedPeople.length === 0 ? (
+        <p>There are no people matching the current search criteria</p>
+      ) : (
+        <table
+          data-cy="peopleTable"
+          className="table is-striped is-hoverable is-narrow is-fullwidth"
+        >
+          <thead>
+            <tr>
+              <th>
+                <span className="is-flex is-flex-wrap-nowrap">
+                  Name
+                  <SearchLink
+                    className="is-active"
+                    params={setSortParam('name')}
+                  >
+                    <span className="icon">
+                      <i className={setSortColumnClasses('name')} />
+                    </span>
+                  </SearchLink>
+                </span>
+              </th>
+
+              <th>
+                <span className="is-flex is-flex-wrap-nowrap">
+                  Sex
+                  <SearchLink
+                    className="is-active"
+                    params={setSortParam('sex')}
+                  >
+                    <span className="icon">
+                      <i className={setSortColumnClasses('sex')} />
+                    </span>
+                  </SearchLink>
+                </span>
+              </th>
+
+              <th>
+                <span className="is-flex is-flex-wrap-nowrap">
+                  Born
+                  <SearchLink
+                    className="is-active"
+                    params={setSortParam('born')}
+                  >
+                    <span className="icon">
+                      <i className={setSortColumnClasses('born')} />
+                    </span>
+                  </SearchLink>
+                </span>
+              </th>
```

Comment:
same here. I'd prefer mapping to avoid code duplications

Reply:
I mapped fields for `th`.